File: libs/destiny/术数.php
<?php

class 术数 {
	private $m_index;
	private $m_element;
	private $m_yinYang;

	public function __construct($element, $yinYang, $index = -1) {
		$this->m_element = $element;
		$this->m_yinYang = $yinYang;
		$this->m_index = $index;
		
		if($index != -1)
			$this->m_yinYang = (abs($index % 2) == 1) ? 阴阳::$阳 : 阴阳::$阴;
	}

	public function getIndex() {
		return $this->m_index;
	}

	public function 五行() {
		return $this->m_element;
	}

	public function 阴阳() {
		return $this->m_yinYang;
	}

	public function equals($obj) {
		if (!($obj instanceof 术数))
			return false;

		$ret = ($this->m_element === $obj->m_element && $this->m_yinYang === $obj->yinYang);
		if ($this->m_index >= 0 && $obj->m_index >= 0)
			$ret = $ret && ($this->m_index == $obj->m_index);

		return ret;
	}
}

File: src/com/infoecos/cn/destiny/lib/术数.java
package com.infoecos.cn.destiny.lib;

import com.infoecos.cn.destiny.lib.common.阴阳;
import com.infoecos.cn.destiny.lib.五行.五行;


public class 术数 {
	private int index = -1;

	private 五行 element;
	private 阴阳 yinYang;

	public 术数(五行 element, 阴阳 yinYang) {
		this.element = element;
		this.yinYang = yinYang;
	}

	public 术数(int index, 五行 element) {
		this.index = index;
		this.element = element;
		yinYang = (Math.abs(index % 2) == 1) ? 阴阳.阳 : 阴阳.阴;
	}

	public int getIndex() {
		return index;
	}

	public 五行 五行() {
		return element;
	}

	public 阴阳 阴阳() {
		return yinYang;
	}

	@Override
	public boolean equals(Object obj) {
		if (!(obj instanceof 术数))
			return false;
		术数 shu = (术数) obj;

		boolean ret = element.equals(shu.element)
				&& yinYang.equals(shu.yinYang);
		if (index >= 0 && shu.index >= 0)
			ret = ret && (index == shu.index);

		return ret;
	}
}
